app/controllers/purchases_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PurchasesController < ApplicationController
   def create
-    if purchase_params[:gateway] == 'paypal'
+    if valid_gateway?
```

Comment:
Considere usar `unless` semelhante a https://github.com/JRRGomes/refactoring-exercise/blob/159b4bf70c297600d5187fcacfaec038e7eff41c/app/controllers/purchases_controller.rb#L8 assim você mantém o `render json` do erro `Gateway not supported!` próximo à verificação da condição.